docs/tools/testing/icetea.md

```diff
@@ -0,0 +1,188 @@
+The interface between the test framework and `DUT` can be
+for example UART or stdio.
+
+More detailed documentation on the tool is available
+[here in rst format](https://github.com/ARMmbed/icetea/tree/master/doc-source)
+and [here in markdown format](https://github.com/ARMmbed/icetea/tree/master/doc).
+
+### Prerequisites
+Icetea supports Linux (Ubuntu preferred), Windows and OS X. Our main target is Linux.
+We support both Python 2.7 and 3.5 or later. Some OS specific prerequisites below:
+
+* Linux
+    * python-dev and python-lxml
+        `sudo apt-get install python-dev python-lxml`
+    * In order to run test cases with hardware in Linux without sudo rights:
+        ```
+        sudo usermod -a -G dialout username
+        Log out & log in back to Linux
+        ```
+        This command will add the user 'username' to the 'dialout' group and
+        grant the required permissions to the USB ports.
+* OS X
+    * [XCode developer tools](http://osxdaily.com/2014/02/12/install-command-line-tools-mac-os-x/)
+    * [MacPorts](https://www.macports.org/install.php)
+    * [lxml](http://lxml.de/installation.html#installation):
+        `STATIC_DEPS=true sudo pip install lxml`
+
+* Windows
+    * python-lxml installation is problematic on Windows since
+    it usually requires build tools. It can however be installed
+    from pre-built binaries.
+        * Search for a binary for you system from the internet.
+        * Navigate to the directory where you downloaded the
+        binary and install it with `pip install <insert_file_name>`
+    * You can follow instructions [here](http://lxml.de/installation.html#installation)
+    instead.
+
+#### Optional
+
+* If you wish to decorate your console log with all kinds of colors,
+install the coloredlogs module using pip. `pip install coloredlogs`
+    * There have been issues with coloredlogs installation on Windows.
+     We might switch to a different module at some point to enable
+     colored logging on Windows as well.
+
+### Installation
+
+`> pip install icetea`
+
+### Usage
+
+To print the help page:
+
+`icetea --help`
+
+To list all local testcases from the examples subfolder:
+
+`icetea --list --tcdir examples`
+
+To print Icetea version:
+
+`icetea --version`
+
+#### Typical use
+
+All of the commands described below might also need other options,
+depending on the test case.
+
+**Running test cases using the tc argument**
+
+`> icetea --tc <test case name> --tcdir <test case search path>`
+
+To run all existing test cases from the `examples` folder:
+
+`> icetea --tc all --tcdir examples`
+
+**Running an example test case with hardware**
+
+In this example, we assume that a compatible board has been connected
+to the computer and an application binary for the board is available.
+The referred test case is available in [the icetea github repository](https://github.com/ARMmbed/icetea/blob/master/examples/test_cmdline.py).
```

Comment:
What do you mean by 'referred test case'?  Is this the example below?  If so, it would be better to say: "The test case below is also available in [the Icetea github repository]